### PR TITLE
Feat/display all current leagues

### PR DESCRIPTION
**Pull request by AtanasKrondev**
- Fetch all current leagues
- Downside: there are too many requests and not all leagues infos are fetched

### CODE REVIEW

File: app/src/lib/fetch-data.ts [hunk 1]
```diff
@@ -107,6 +107,15 @@ const transformScore = (scores: any, sport: Sport) => {
   }
 }
 
+export const fetchCurrentLeaguesIds = async (sport: Sport) => {
+  const res = await fetchSportData(sport, '/leagues', new URLSearchParams())
```
Comment (imollov) [inline app/src/lib/fetch-data.ts]: To address your concern about the number of requests caused here - maybe we can pass a parameter to revalidate the cache only once per day.

File: app/src/app/page.tsx [hunk 2]
```diff
@@ -14,8 +15,9 @@ export default async function Home({
 }: {
   searchParams: { [key: string]: string | string[] | undefined }
 }) {
+  const currentLeaguesIds = await fetchCurrentLeaguesIds(Sport.Rugby)
   const data = await Promise.all(
-    leagueIds[Sport.Rugby].map(async (leagueId) => {
```
Comment (imollov) [inline app/src/app/page.tsx:18]: `leagueIds` is unused after this change and can be removed everywhere.

**Comment by imollov** [comment]
There seems to be an issue with the league logos. Sometimes there's no source for the image and sometimes is not sized correctly.

Screenshot 2023-10-24 at 10 07 31

Screenshot 2023-10-24 at 10 07 34

File: app/src/lib/fetch-data.ts (hunk 1 above)
Comment (imollov) [reply app/src/lib/fetch-data.ts]: It should actually be on the league details request, but I can't put a comment there.

Comment (imollov) [reply app/src/lib/fetch-data.ts]: It also makes sense to have it here, though.

**Comment by AtanasKrondev** [comment]
> There seems to be an issue with the league logos. Sometimes there's no source for the image and sometimes is not sized correctly.
> 

The missing logos are due to 
``
  errors: {
    rateLimit: 'Too many requests. You have exceeded the limit of requests per minute of your subscription.'
  },
``

As a workaround we can display the missing image placeholder, that the API returns for some of the logos anyway 🤷🏻‍♂️
image

**Comment by imollov** [comment]
> > There seems to be an issue with the league logos. Sometimes there's no source for the image and sometimes is not sized correctly.
> 
> The missing logos are due to `errors: { rateLimit: 'Too many requests. You have exceeded the limit of requests per minute of your subscription.' },`
> 
> As a workaround we can display the missing image placeholder, that the API returns for some of the logos anyway 🤷🏻‍♂️ image

OK, let's do that. 

As a separate solution, we can implement pagination/load more functionality and not fetch all league games at once (https://github.com/hackbg/prediction-game/issues/2).

**Pull request closed without merging**